### PR TITLE
Kabirwalia/add automl object detection

### PR DESCRIPTION
**Pull request by kabirwalia8300**
Adds DRISE support for automl mlflow object detection models. This PR is a dependency for the ml-wrappers PR https://github.com/microsoft/ml-wrappers/pull/113.

### CODE REVIEW

File: python/vision_explanation_methods/DRISE_runner.py [hunk 1]
```diff
@@ -130,24 +134,52 @@ def get_drise_saliency_map(
...
-        verbose=True  # Turns progress bar on/off.
-    )
+    if isinstance(model, MLflowDRiseWrapper):
+        x, y = test_image.size
+        imgio = BytesIO()
+        test_image.save(imgio, format='PNG')
+        img_str = base64.b64encode(imgio.getvalue()).decode('utf8')
+        img_input = pd.DataFrame(
+            data=[[img_str, (y, x)]],
+            columns=['image', "image_size"],
```
Comment (Advitya17) [inline python/vision_explanation_methods/DRISE_runner.py:144]: Nit: single v/s double quotes

File: python/vision_explanation_methods/DRISE_runner.py [hunk 2]
```diff
@@ -130,24 +134,52 @@ def get_drise_saliency_map(
...
+            target_detections=detections,
+            # This is how many masks to run -
+            # more is slower but gives higher quality mask.
+            number_of_masks=nummasks,
+            mask_padding=maskpadding,
+            device=device,
+            # This is the resolution of the random masks.
+            # High resolutions will give finer masks, but more need to be run.
+            mask_res=maskres,
+            verbose=True  # Turns progress bar on/off.
```
Comment (Advitya17) [inline python/vision_explanation_methods/DRISE_runner.py:161]: Considering `detections` and `saliency_scores` are both the last variables in the if-else conditions, should prolly push these out of the conditions.

File: python/vision_explanation_methods/explanations/drise.py [hunk 3]
```diff
@@ -257,3 +260,118 @@ def DRISE_saliency(
...
+            masked_image)
+
+        masked_df = pd.DataFrame(
+            data=[[masked_image_str, masked_image_size]],
+            columns=['image', "image_size"],
+        )
+
+        masked_detections = model.predict(masked_df)
+
+        affinity_scores = []
```
Comment (Advitya17) [inline python/vision_explanation_methods/explanations/drise.py:367]: Consider list comprehension for faster execution with the python interpreter.

File: python/vision_explanation_methods/explanations/drise.py [hunk 4]
```diff
@@ -257,3 +260,118 @@ def DRISE_saliency(
...
+        for (target_detection, masked_detection) in zip(target_detections,
+                                                        masked_detections):
+            affinity_scores.append(
+                compute_affinity_scores(target_detection, masked_detection))
+
+        mask_records.append(MaskAffinityRecord(
+            mask=mask.to("cpu"),
+            affinity_scores=[s.detach().to("cpu") for s in affinity_scores])
```
Comment (Advitya17) [inline python/vision_explanation_methods/explanations/drise.py:375]:
```suggestion
            affinity_scores=[s.detach().cpu() for s in affinity_scores])
```
Nit: avoid hardcoded str value.

File: python/vision_explanation_methods/explanations/drise.py [hunk 5]
```diff
@@ -257,3 +260,118 @@ def DRISE_saliency(
...
+        masked_detections = model.predict(masked_df)
+
+        affinity_scores = []
+        for (target_detection, masked_detection) in zip(target_detections,
+                                                        masked_detections):
+            affinity_scores.append(
+                compute_affinity_scores(target_detection, masked_detection))
+
+        mask_records.append(MaskAffinityRecord(
+            mask=mask.to("cpu"),
```
Comment (Advitya17) [inline python/vision_explanation_methods/explanations/drise.py:374]:
```suggestion
            mask=mask.detach().cpu(),
```

**Review by Advitya17 — APPROVED** [review]

File: python/vision_explanation_methods/explanations/drise.py (hunk 4 above)
Comment (kabirwalia8300) [reply python/vision_explanation_methods/explanations/drise.py:372]: fixed

File: python/vision_explanation_methods/explanations/drise.py (hunk 5 above)
Comment (kabirwalia8300) [reply python/vision_explanation_methods/explanations/drise.py:371]: fixed

File: python/vision_explanation_methods/explanations/drise.py (hunk 3 above)
Comment (kabirwalia8300) [reply python/vision_explanation_methods/explanations/drise.py:364]: fixed

File: python/vision_explanation_methods/DRISE_runner.py (hunk 1 above)
Comment (kabirwalia8300) [reply python/vision_explanation_methods/DRISE_runner.py:143]: fixed

File: python/setup.py [hunk 6]
```diff
@@ -41,7 +41,9 @@
     'tqdm',
     # TODO: remove this dependency
     'matplotlib<3.7.0',
-    'ml_wrappers'
+    # 'ml_wrappers'
+    'ml_wrappers @ https://publictestdatasets.blob.core.windows.net/packages/'\
```
Comment (imatiach-msft) [inline python/setup.py:45]: let's do a release of ml-wrappers first then before merging this PR

**Review by imatiach-msft — APPROVED** [review]

**Pull request merged**